### PR TITLE
[buteo-sync-plugins-social] Fix exdate parsing in Google Calendar sync

### PR DESCRIPTION
**Pull request by chriadam**
This commit fixes exdate parsing by ensuring that the string is
properly trimmed prior to converting from string to datetime,
and allowing for different datetime formats in case other client
applications sync with different formats.

### CODE REVIEW

File: src/google/google-calendars/googlecalendarsyncadaptor.cpp [hunk 1]
```diff
@@ -163,11 +172,25 @@ QList<KDateTime> datetimesFromExRDateStr(const QString &exrdatestr, bool *isDate
             if (dtstr.endsWith('Z')) {
                 // UTC
                 KDateTime kdt = KDateTime::fromString(dtstr, RFC5545_KDATETIME_FORMAT);
+                if (!kdt.isValid()) {
+                    // try parsing from alternate formats
+                    kdt = KDateTime::fromString(dtstr, RFC3339_FORMAT);
+                }
+                if (!kdt.isValid()) {
+                    SOCIALD_LOG_ERROR("unable to parse datetime from ex/rdate string:" << exrdatestr);
+                }
```
Comment (chriadam) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: this should be an else block instead

File: src/google/google-calendars/googlecalendarsyncadaptor.cpp [hunk 2]
```diff
@@ -163,11 +172,25 @@ QList<KDateTime> datetimesFromExRDateStr(const QString &exrdatestr, bool *isDate
...
             } else {
                 // Floating time
                 KDateTime kdt = KDateTime::fromString(dtstr, RFC5545_KDATETIME_FORMAT_NTZC);
+                if (!kdt.isValid()) {
+                    // try parsing from alternate formats
+                    kdt = KDateTime::fromString(dtstr, RFC3339_FORMAT_NTZC);
+                }
+                if (!kdt.isValid()) {
+                    SOCIALD_LOG_ERROR("unable to parse datetime from ex/rdate string:" << exrdatestr);
+                }
```
Comment (chriadam) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: ditto, else

File: src/google/google-calendars/googlecalendarsyncadaptor.cpp [hunk 3]
```diff
@@ -1248,6 +1271,10 @@ void GoogleCalendarSyncAdaptor::upsyncFinishedHandler()
                     SOCIALD_LOG_DEBUG("  old start:" << oldDTS << ", old end:" << oldDTE);
                     SOCIALD_LOG_DEBUG("  new start:" << event->dtStart().toString(RFC3339_FORMAT) <<
                                       ", new end:" << event->dtEnd().toString(RFC3339_FORMAT));
+                    SOCIALD_LOG_DEBUG("  exdates:");
```
Comment (chriadam) [inline src/google/google-calendars/googlecalendarsyncadaptor.cpp]: I'm not certain whether to remove these log lines or not.

**Comment by pvuorela** [comment]
LGTM

**Pull request merged**